src/Nri/Ui/Checkbox/V6.elm

```diff
@@ -0,0 +1,689 @@
+        , Attributes.id model.identifier
+        , Attributes.disabled model.disabled
+        ]
+
+
+viewEnabledLabel :
+    { a
+        | identifier : String
+        , setterMsg : Bool -> msg
+        , selected : IsSelected
+        , label : String
+    }
+    -> (String -> Html.Html msg)
+    -> Maybe Css.Color
+    -> Icon
+    -> Html.Html msg
+viewEnabledLabel model labelView backgroundColor icon =
+    Html.Styled.label
+        [ Attributes.for model.identifier
+        , Aria.controls model.identifier
+        , Widget.disabled False
+        , Widget.checked (selectedToMaybe model.selected)
+        , Attributes.tabindex 0
+        , HtmlExtra.onKeyUp
+            { defaultOptions | preventDefault = True }
+            (\keyCode ->
+                -- 32 is the space bar, 13 is enter
+                if keyCode == 32 || keyCode == 13 then
+                    Just (onCheck model)
+
+                else
+                    Nothing
+            )
+        , labelClass model.selected
+        , css
+            [ positioning
+            , textStyle
+            , outline none
+            , cursor pointer
+            , backgroundCss backgroundColor
+            ]
+        ]
+        [ viewIcon [] icon
+        , labelView model.label
+        ]
+
+
+onCheck : { a | setterMsg : Bool -> msg, selected : IsSelected } -> msg
+onCheck model =
+    selectedToMaybe model.selected
+        |> Maybe.withDefault False
+        |> not
+        |> model.setterMsg
+
+
+viewDisabledLabel :
+    { a | identifier : String, selected : IsSelected, label : String }
+    -> (String -> Html.Html msg)
+    -> Icon
+    -> Html.Html msg
+viewDisabledLabel model labelView icon =
+    Html.Styled.label
+        [ Attributes.for model.identifier
+        , Aria.controls model.identifier
+        , Widget.disabled True
+        , Widget.checked (selectedToMaybe model.selected)
+        , labelClass model.selected
+        , css
+            [ positioning
+            , textStyle
+            , outline none
+            , cursor auto
+            ]
+        ]
+        [ viewIcon [ opacity (num 0.4) ] icon
+        , labelView model.label
+        ]
+
+
+labelClass : IsSelected -> Html.Styled.Attribute msg
+labelClass isSelected =
+    case isSelected of
+        Selected ->
+            toClassList [ "Label", "Checked" ]
+
+        NotSelected ->
+            toClassList [ "Label", "Unchecked" ]
+
+        PartiallySelected ->
+            toClassList [ "Label", "Indeterminate" ]
+
+
+toClassList : List String -> Html.Styled.Attribute msg
+toClassList =
+    List.map (\a -> ( "checkbox-V6__" ++ a, True )) >> Attributes.classList
+
+
+backgroundCss : Maybe Css.Color -> Style
+backgroundCss bgColor =
+    case bgColor of
+        Nothing ->
+            Css.empty []
+
+        Just color ->
+            backgroundColor color
+
+
+positioning : Style
+positioning =
+    batch
+        [ display inlineBlock
+        , padding4 (px 13) (px 10) (px 13) (px 45)
```

Comment:
added 10px to the left/right so that the highlight frames the section more elegantly.